### PR TITLE
Dockefile for Cling

### PR DESCRIPTION
**Pull request by unlisted**
Cling is an interactive C++ interpreter, built on top of Clang and LLVM compiler infrastructure. The binary distribution includes Jupyter kernels for c++11, c++14 and c++17. This Dockerfile downloads binary distribution and installs it to /opt, and installs the Jupyter kernels.

### CODE REVIEW

**Comment by jakirkham** [comment]
In conda-forge, @SylvainCorlay and others built `cling`. Maybe we want to use it here as well?

File: cling-notebook/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+RUN cd /tmp && \
+    wget --quiet https://root.cern.ch/download/cling/${CLING_VERSION}.tar.bz2 && \
+    tar -C /opt -xjf ${CLING_VERSION}.tar.bz2 && \
+    cd /opt/${CLING_VERSION}/share/cling/Jupyter/kernel && \
+    pip3 install -e . && \
+    jupyter-kernelspec install cling-cpp17 && \
+    jupyter-kernelspec install cling-cpp1z && \
+    jupyter-kernelspec install cling-cpp14 && \
+    jupyter-kernelspec install cling-cpp11
```
Comment (dalg24) [inline cling-notebook/Dockerfile]: You might want to do `rm /tmp/${CLING_VERSION}.tar.bz2`

**Comment by parente** [comment]
Hi @unlisted. We'd like to start encouraging the growth of community-maintained stacks in other repositories as an alternative to continually expanding the set and scope of images maintained here. I think this image would be an excellent start to the ecosystem we're trying to cultivate, and would love it if you were to be one of the first to follow the new [community stacks documentation](http://jupyter-docker-stacks.readthedocs.io/en/latest/contributing/stacks.html) to set up a project.

If you do decide to give it a shot, please feel free to submit a PR updating the documentation site with a link to the git repository. (The community stacks documentation includes this suggestion along with steps to do it.)

**Comment by parente** [comment]
I'm going to close this PR as inactive. If someone picks up the torch and creates a community maintained Cling image, please submit a PR to linking to it from the documentation.

**Pull request closed without merging**